persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/jdbc/JdbcPersistenceComponents.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.persistence.jdbc
+
+import scala.concurrent.ExecutionContext
+
+import akka.actor.ActorSystem
+import com.lightbend.lagom.internal.persistence.jdbc.SlickProvider
+import com.lightbend.lagom.internal.scaladsl.persistence.ReadSideConfig
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcOffsetStore
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcReadSideImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.OffsetTableConfiguration
+import play.api.Configuration
+import play.api.db.DBApi
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcPersistentEntityRegistry
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntityRegistry
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcSessionImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.OffsetStore
+
+/**
+ * Persistence JDBC components (for compile-time injection).
+ */
+trait JdbcPersistenceComponents {
+
+  def system: ActorSystem
+  def executionContext: ExecutionContext
+  def configuration: Configuration
+  def dbApi: DBApi // FIXME where is this created?
+  def readSideConfig: ReadSideConfig
```

Comment:
@jroper I was thinking if I should also make it extend `PersistenceComponents`. Then the following 3 dependencies would come from there: `actorSystem`, `executionContext `, `configuration`.

Thinking some more about how these modules can be used it is really two things; write-side and read-side. Perhaps they should be defined in different *Components traits so that a service can only use the write-side. I'll see if that separation makes sense.